core/src/thrift/tachyon.thrift

```diff
@@ -257,23 +287,69 @@ service WorkerService {
   bool asyncCheckpoint(1: i32 fileId)
     throws (1: TachyonException e)
 
+  /**
+   * Used to cache a block into Tachyon's cache space, worker will move the temporary block file
+   * from user folder to data folder, and update the space usage information related. then update
+   * the block information to master. 
+   */
   void cacheBlock(1: i64 userId, 2: i64 blockId)
     throws (1: FileDoesNotExistException eP, 2: SuspectedFileSizeException eS,
       3: BlockInfoException eB)
 
-  string getDataFolder()
+  /**
+   * Used to cancel a block which is being written. worker will delete the temporary block file and
+   * the location and space information related, then reclaim space allocated to the block.
+   */
+  void cancelBlock(1: i64 userId, 2: i64 blockId)
 
-  string getUserTempFolder(1: i64 userId)
+  /**
+   * Used to allocate location and space for a new coming block, worker will choose the appropriate
+   * storage directory which fits the initial block size by some allocation strategy, and the
+   * temporary file path of the block file will be returned. if there is no enough space on Tachyon
+   * storage OutOfSpaceException will be thrown, if the file is already being written by the user,
+   * FileAlreadyExistException will be thrown.
+   */
+  string getBlockLocation(1: i64 userId, 2: i64 blockId, 3: i64 initialBytes)
+    throws (1: OutOfSpaceException eP, 2: FileAlreadyExistException eS)
 
+  /**
+   * Used to get user's temporary folder on under file system, and the path of the user's temporary
+   * folder will be returned.
+   */
   string getUserUfsTempFolder(1: i64 userId)
 
-  void lockBlock(1: i64 blockId, 2: i64 userId) // Lock the file in memory while the user is reading it.
+  /**
+   * Lock the file in Tachyon's cache space while the user is reading it, and the path of the block
+   * file locked will be returned, if the block file is not found, FileDoesNotExistException will
+   * be thrown.
+   */
+  string lockBlock(1: i64 blockId, 2: i64 userId)
+    throws (1: FileDoesNotExistException eP)
 
-  void returnSpace(1: i64 userId, 2: i64 returnedBytes)
+  /**
+   * Used to promote block on under storage layer to top storage layer when there are more than one
+   * storage layers in Tachyon's cache space. return true if the block is successfully promoted,
```

Comment:
"Used to promote block on under storage layer to top storage layer when there are more than one storage layers in Tachyon's cache space."

What if there is only one layer in Tachyon?